### DISCUSSION TITLE
Any good SDF reader/converter  for Windows?

### DISCUSSION THREAD

**Discussion by DanRight** [epochpic/epoch#373] [General]
Is there any (ideally command prompt based) reader/converter of SDF format files into binary/text files like the h5dump tool for HDF5 files for example?

**Comment by TomGoffrey** [comment]
Have you tried using `sdffilter` in `epoch/SDF/utilities`?

**Comment by DanRight** [comment]
Thanks Tom for the hint. 

In Linux (i do not have any compiler for Windows which compiles this code) I needed to make a change in Make file for Intel compilers
MPIF90=mpiifort
(same change was required to compile EPOCH) otherwise it does not even start compilation. But despite of that still no sdffilter executable file was produced ;(

**Reply by TomGoffrey** [reply]
How exactly are you trying to build the tool? I'm not sure why you think you need an MPI or Fortran compile for the sdffilter tool.

For EPOCH you don't need to modify the Makefile, you can specify `MPIF90=mpiifort` from the command line.

**Reply by DanRight** [reply]
Again thanks for the respond.
In the SDF/Fortran folder there is a Makefile which has definition 
MPIF90 ?= mpif90

And this Makefile is set by default for using pgi compiler. If you just edit this line to COMPILER=intel this Makefile does not work. I do not know if this is because latest Intel Fortran compiler changed that requiring MPIF90=mpiifort instead of MPIF90 = mpif90 or our computer Linux setup changed that.  But this is definitely confusing part.

(Similarly with EPOCH code own Makefile. The description in Help has confusing and wrong statement that Make by default using Intel compiler and you do not need to edit anything, just type Make and run - this caused me to lose 1 week because after fail i tried all other options like gfortran and then again Intel etc and that additionally screws newbie messing up libraries which need purge before using another compiler as i found later in other places )

After running Make compilation starts and after a minute or so ends without errors. But at the end there is no executable generated anywhere

I am trying to get SDF working in desperation to see the EPOCH run results. I essentially do not need any SDF outputs of EPOCH, just to get particle and field plain ASCII/binary data file is all what i need. But the code is extremely difficult to read and modify without description of variables and at least short explanation how for example output  works. 

What i need is just to open the file and drop there all coordinates, momentum, weight, ion charge and ID for the particles and similar file for the fields. That is 10 lines of Fortran code instead of 800k of SDF related source code with tons of CLASSes, TYPEs, POINTERs i am trying to decipher for a month or even more :). 99% of Fortran users have very limited experience with that new stuff. The SDF approach is probably useful for compatibility with different computers of different systems but just the unformatted file output is enough for me being similarly fast or even faster than any SDFs or HDF5s. Around 8 GB per second is what PCIe4 M.2 drives can deliver and the plain unformatted I/O gets that. And there were no problems reading unformatted outputs from Linux computers even on Windows PCs, without any modifications. If somebody here helped me to pass this hard early learning curve then all after that will be much easier.

**Comment by DanRight** [comment]
Error in Intel  Make?

**Comment by DanRight** [comment]
Anyone has sdffilter  executable for Windows to share? I think it is good idea if authors  prepared such handy tool for more mass use. Like the tools for HDF5 for example which exist for multiple platforms in EXE form. 

It is difficult to compile sdffilter in Linux from current sources as this is really heavily multilanguage project needing besides Fortran, also C and Python
